### DISCUSSION TITLE
Trouble interfacing with I2C load cell

### DISCUSSION THREAD

**Discussion by Kibstar** [micropython/micropython#11332] [RP2040 / Pico] [answered]
Hi all,

I’m having some trouble interfacing with a load cell that I recently bought for a project. It uses the I2C protocol to send data from the cell. I’ve been reading the data sheet and it looks like I need to send a read request to the I2C bus and then read the following two bytes. I’m having a lot of trouble trying to send the read request. I’ve searched the internet for hours but had no luck finding something that applies to my situation. Any code examples or suggestions are welcome.

Data sheet here: (page 10) https://www.te.com/commerce/DocumentDelivery/DDEController?Action=showdoc&DocId=Data+Sheet%7FFX29%7FA3%7Fpdf%7FEnglish%7FENG_DS_FX29_A3.pdf

I have followed the data sheet correctly and I have the correct pull up resistors and capacitors in the correct locations

Load cell model: FX29K0-100A-0025-L

Code I am trying to run:

```
import machine, time

scl_pin = machine.Pin(15)
sda_pin = machine.Pin(14)

address = 0x28

READ_REG = 6

def force_data(data):
    value = data[0] << 8 | data[1]
    force = (value & 0xFFF)
    if value & 0x1000:
        force += 4096
    return force


i2c = machine.I2C(1, scl=scl_pin, sda=sda_pin, freq=400000)

i2c.writeto_mem(address, 0, bytes([0]))

while True:
    
    data = i2c.readfrom_mem(address, READ_REG, 2)
    print(force_data(data))
    time.sleep(.04)

```

I do not receive any error message, but instead I get a constant output of 8191 from the load cell. The load cell does not react to any force applied

Thanks in advance.

**Comment by robert-hh** [comment]
As far as I understand the data sheet, you have to send a read request with 0 data to wake up the device and start measurement,wait at least 8.4 ms, and the a second read request for 2 bytes of data.

**Reply by Kibstar** [reply]
Do you know what that might look like in code? - Here's my attempt (with no luck):

```
import machine, time

scl_pin = machine.Pin(15)
sda_pin = machine.Pin(14)

address = 0x28

READ_REG = 6

def force_data(data):
    value = data[0] << 8 | data[1]
    force = (value & 0xFFF)
    if value & 0x1000:
        force += 4096
    return force

i2c = machine.I2C(1, scl=scl_pin, sda=sda_pin, freq=400000)


while True:
    i2c.readfrom(address, READ_REG, 0)
    data = i2c.readfrom_mem(address, READ_REG, 2)
    print(force_data(data))
    time.sleep(.04)
```

**Reply by robert-hh** [reply, marked as answer]
A delay should be between the first and second read, e.g.:
```
while True:
    i2c.readfrom(address, READ_REG, 0)
    time.sleep_ms(10)
    data = i2c.readfrom_mem(address, READ_REG, 2)
    print(force_data(data))
    time.sleep(1)
```

**Reply by Kibstar** [reply]
It works! Amazing stuff, Robert! Thank you!

**Reply by robert-hh** [reply]
And thank you very much for pointing me at this load cell sensor. That device looks useful.
